Tests/MongoSwiftTests/AsyncAwaitTestUtils.swift

```diff
@@ -51,6 +51,25 @@ func assertIsEventuallyTrue(
     XCTFail("Expected condition \"\(description)\" to be true within \(timeout) seconds, but was not")
 }
 
+/// Asserts that the provided block returns true within the specified timeout. Nimble's `toEventually` can only be used
+/// rom the main testing thread which is too restrictive for our purposes testing the async/await APIs.
+@available(macOS 10.15.0, *)
+func assertIsEventuallyTrue(
+    description: String,
+    timeout: TimeInterval = 5,
+    sleepInterval: TimeInterval = 0.1,
+    _ block: () async throws -> Bool
+) async throws {
+    let start = DispatchTime.now()
```

Comment:
Given that our block is now async and could potentially run for a while, I think we'll want to try to cancel it when the deadline has been hit rather than checking if we've timed out after each time it gets executed.

To do that, I think we'll need to run the work in a Task and cancel that task after a certain amount of time. To do the cancellation part, I think we'll need another task that will sleep for the timeout and then cancel the other task afterwards.

e.g.

``` swift
// create a task that performs the async closure in a loop
let workTask = Task {
    while !Task.isCancelled {
        guard try await block() else {
            try? Task.sleep(nanoseconds: Int64(sleepInterval * 1_000_000_000))
            continue
        }
        return true
    }
    return false
}

// create another task that will cancel the previous one after the timeout is hit
Task {
    try await Task.sleep(nanoseconds: timeout * 1_000_000_000)
    // we'll only reach here if this task doesn't get cancelled
    workTask.cancel()
}
 
guard try await workTask.value else {
    XCTFail("Expected condition...")
}
```

Also there may be an easier way to do this, I'm still new to Swift concurrency myself!

Reply:
Just wanted to make sure I understand this. 

Understand that we don't to start a potentially long task after checking we are in bounds. This sounds similar to the all hands talk when the driver detects that an operation will indeed take too long and shoudlnt bother doing it. 

From my understanding, the unnamed task waits `timeout` seconds and then cancels the `workTask` (which is basically timing out). At the same time, the `workTask` attempts to do the work and sleep/retry if not until it is cancelled. If the work does indeed complete, we return true and false if not. 

In terms of simpler ways, this is pretty logical (with comments). The only other way I can think of is not using the `workTask` as a task and using a while loop similar to the original and using an atomic bool to basically return false when we run out of time (set in the other task in place of `workTask.cancel`. However, this seems like it would be hard to implement because we run into the original issue of potentially doing a long task just before timeout time. 

TL;DR: Implemented in a similar way as recommended